subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/TestResult.java

```diff
@@ -46,6 +48,16 @@
     @Nullable
     Throwable getException();
 
+    /**
+     * If the test failed with any exceptions, this will contain the exceptions and a description of the failure types.
+     *
+     * @return The failures, if any, logged for this test.  If none, an empty list is returned.
+     * @see #getExceptions()
+     * @since 7.5
+     */
+    @Incubating
+    List<TestFailure> getFailures();
```

Comment:
I've put the new method here only to follow the control-flow of `getExceptions`. Maybe we should push this method down to `DefaultTestResut` and do a downcast at the callsite.